library/pkparse.c

```diff
@@ -1455,8 +1455,10 @@ int mbedtls_pk_parse_public_key( mbedtls_pk_context *ctx,
         if( ( pk_info = mbedtls_pk_info_from_type( MBEDTLS_PK_RSA ) ) == NULL )
```

Comment:
If this fails, the PEM buffer leaks. I don't think this _can_ fail though — if `MBEDTLS_RSA_C` is enabled then `mbedtls_pk_info_from_type` returns a non-null pointer. So why are we testing this condition, but not handling it correctly? I'd rather either remove the test or ensure that the PEM buffer is freed on this path.

Instead of spreading `mbedtls_pem_free( &pem );` all over the place, I think we should switch this function to a ingle return with `goto exit` and `exit: mbedtls_pem_free( &pem ); return( ret );`.